lib/flutter_map_tappable_polyline.dart

```diff
@@ -99,28 +102,67 @@ class TappablePolylineLayer extends StatelessWidget {
-                TaggedPolyline polyline = polylineOpts.polylines.firstWhere(
-                    (TaggedPolyline polylineOpt) => polylineOpt.offsets.firstWhere(
-                        (Offset offset) =>
-                            (offset.dx / polylineOpts.pointerDistanceTolerance)
-                                        .round()
-                                        .toDouble() *
-                                    polylineOpts.pointerDistanceTolerance ==
-                                (details.localPosition.dx / polylineOpts.pointerDistanceTolerance)
-                                        .round() *
-                                    polylineOpts.pointerDistanceTolerance &&
-                            (offset.dy / polylineOpts.pointerDistanceTolerance)
-                                        .round()
-                                        .toDouble() *
-                                    polylineOpts.pointerDistanceTolerance ==
-                                (details.localPosition.dy /
-                                            polylineOpts.pointerDistanceTolerance)
-                                        .round() *
-                                    polylineOpts.pointerDistanceTolerance,
-                        orElse: () => null) is Offset,
-                    orElse: () => null);
-
-                if (polyline is TaggedPolyline) onTap(polyline);
+                // Calculating taps in between points on the polyline. We
+                // iterate over all the segments in the polyline to find any
+                // matches with the tapped point within the
+                // pointerDistanceTolerance.
+                for (var i = 0; i < polylineOpts.polylines.length; i++) {
+                  Polyline currentPolyline = polylineOpts.polylines[i];
+
+                  for (var j = 0; j < currentPolyline.offsets.length - 1; j++) {
+                    // We consider the points point1, point2 and tap points in a triangle
+                    var point1 = currentPolyline.offsets[j];
+                    var point2 = currentPolyline.offsets[j + 1];
+                    var tap = details.localPosition;
+
+                    // To determine if we have tapped in between two points, we
+                    // calculate the length from the tapped point to the line
+                    // created by point1, point2. If this distance is shorter
+                    // than the specified threshold, we have detected a tap
+                    // between two points.
+                    //
+                    // We start by calculating the length of all the sides using pythagoras.
+                    var a = distance(point1, point2);
+                    var b = distance(point1, tap);
+                    var c = distance(point2, tap);
+
+                    // To find the height when we only know the lengths of the sides, we can use Herons formula to get the Area.
+                    var semiPerimeter = (a + b + c) / 2.0;
```

Comment:
Is `a - b - c <= polylineOpts.pointerDistanceTolenrance` sufficient?

Reply:
I tested it using this formula, but there were some edge-cases that would not be covered properly.

Reply:
OK. LGTM